<p>If a <code>private</code> field is declared but not used in the program, it can be considered dead code and should therefore be removed. This will
improve maintainability because developers will not wonder what the variable is used for.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class MyClass {
  private $foo = 4;                       //foo is unused

  public function compute($a) {
    return $a * 4;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyClass {

  public function compute($a) {
    return $a * 4;
  }
}
</pre>

