<p>The PSR2 standard recommends listing modifiers in the following order to improve the readability of PHP source code:</p>
<ol>
  <li> final or abstract </li>
  <li> public or protected or private </li>
  <li> static </li>
</ol>
<h2>Noncompliant Code Example</h2>
<pre>
static protected $foo;
...
public static final function bar(){...}
</pre>
<h2>Compliant Solution</h2>
<pre>
protected static $foo;
...
final public static function bar(){...}
</pre>

