<p>For better readability, do not put more than one statement on a single line.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if(someCondition) doSomething();
</pre>
<h2>Compliant Solution</h2>
<pre>
if(someCondition) {
  doSomething();
}
</pre>
<h2>Exceptions</h2>
<p>Anonymous functions containing a single statement are ignored.</p>
<pre>
$max_comparator = function ($v) { return $v &gt; 2; };           // Compliant
$max_comparator = function ($v) { echo $v; return $v &gt; 2; };  // Noncompliant
</pre>

