<p>The following predefined variables are deprecated and should be replaced by the new versions:</p>
<table>
  <tbody>
    <tr>
      <th>Replace</th>
      <th>With</th>
    </tr>
    <tr>
      <td>$HTTP_SERVER_VARS</td>
      <td>$_SERVER</td>
    </tr>
    <tr>
      <td>$HTTP_GET_VARS</td>
      <td>$_GET</td>
    </tr>
    <tr>
      <td>$HTTP_POST_VARS</td>
      <td>$_POST</td>
    </tr>
    <tr>
      <td>$HTTP_POST_FILES</td>
      <td>$_FILES</td>
    </tr>
    <tr>
      <td>$HTTP_SESSION_VARS</td>
      <td>$_SESSION</td>
    </tr>
    <tr>
      <td>$HTTP_ENV_VARS</td>
      <td>$_ENV</td>
    </tr>
    <tr>
      <td>$HTTP_COOKIE_VARS</td>
      <td>$_COOKIE</td>
    </tr>
    <tr>
      <td>$php_errormsg</td>
      <td>error_get_last()</td>
    </tr>
  </tbody>
</table>
<h2>Noncompliant Code Example</h2>
<pre>
echo 'Name parameter value: ' . $HTTP_GET_VARS["name"];
</pre>
<h2>Compliant Solution</h2>
<pre>
echo 'Name parameter value: ' . $_GET["name"];
</pre>

