<p>In PHP 5 both the way to declare a constructor and the way to make a call to a parent constructor have evolved. When declaring constructors with
the PHP5 <code>__construct</code> name, nested calls to parent constructors should also use the new <code>__constructor</code> name.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class Foo extends Bar {
  function __construct() {
    parent::Bar();
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
class Foo extends Bar {
  function __construct() {
    parent::__construct();
  }
}
</pre>

