<p>When either the equality operator in a null test or the logical operator that follows it is reversed, the code has the appearance of safely
null-testing the object before dereferencing it. Unfortunately the effect is just the opposite - the object is null-tested and then dereferenced
<em>only</em> if it is null, leading to a guaranteed null pointer dereference.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if ($obj == null &amp;&amp; $obj-&gt;isOpen()) {
  echo "Object is open";
}

if ($obj != null || $obj-&gt;isOpen()) {
  echo "Object is not open";
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if ($obj == null || $obj-&gt;isOpen()) {
  echo "Object is open";
}

if ($obj != null &amp;&amp; !$obj-&gt;isOpen()) {
  echo "Object is not open";
}
</pre>

