<p>The PHP 4 method of declaring a variable, using the <code>var</code> keyword, was deprecated in early versions of PHP 5. Even though it's not
considered deprecated in the most recent versions, it's nonetheless not best practice to use it. When <code>var</code> does appear, it is interpreted
as a synonym for <code>public</code> and treated as such. Therefore <code>public</code> should be used instead.</p>
<p>From the PHP Manual:</p>
<blockquote>
  <p>The PHP 4 method of declaring a variable with the var keyword is still supported for compatibility reasons (as a synonym for the public keyword).
  In PHP 5 before 5.1.3, its usage would generate an E_STRICT warning.</p>
</blockquote>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;?php
class Foo
{
    var $bar = 1;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;?php
class Foo
{
    public $bar = 1;
}
</pre>

