<p>Using indifferently lower or upper case for PHP keywords and constants "true", "false" and "null" can impact the readability of PHP source
code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;?php ECHO 'Hello World'; ?&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;?php echo 'Hello World'; ?&gt;
</pre>

