<p>There is no need to use the <code>final</code> modifier inside a <code>final</code> class. Everything in it is <code>final</code> by default.</p>
<h2>Noncompliant Code Example</h2>
<pre>
final class MyClass {

  public final String getName() {  // Noncompliant
    return name;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
final class MyClass {

  public String getName() {  // Compliant
    return name;
  }
}
</pre>

