<p>PHP has two sets of logical operators: <code>&amp;&amp;</code> / <code>||</code>, and <code>and</code> / <code>or</code>. The difference between
the sets is precedence. Because <code>and</code> / <code>or</code> have a lower precedence than almost any other operator, using them instead of
<code>&amp;&amp;</code> / <code>||</code> may not have the result you expect.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$have_time = true;
$have_money = false;
$take_vacation = $have_time and $have_money;  // Noncompliant. $take_vacation == true.
</pre>
<h2>Compliant Solution</h2>
<pre>
$have_time = true;
$have_money = false;
$take_vacation = $have_time &amp;&amp; $have_money;  // $take_vacation == false.
</pre>

