<p>A <code>catch</code> clause that only rethrows the caught exception has the same effect as omitting the <code>catch</code> altogether and letting
it bubble up automatically, but with more code and the additional detriment of leaving maintainers scratching their heads. </p>
<p>Such clauses should either be eliminated or populated with the appropriate logic.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$s = "";
try {
  $s = readMyFile($fileName);
} catch (Exception $e)  {
  throw $e;  // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
$s = "";
try {
  $s = readMyFile($fileName);
} catch (Exception $e) {
  error_log($e-&gt;getMessage());
  throw $e;  // Compliant
}
</pre>
<p>or</p>
<pre>
$s = "";
try {
  $s = readMyFile($fileName);
} catch (Exception $e) {
  error_log($e-&gt;getMessage());
  throw new MyException("an exception occurred", 2, $e);  // Compliant
}
</pre>
<p>or</p>
<pre>
$s = readMyFile($fileName);
</pre>

