<p>Dependency injection is a software design pattern in which one or more dependencies (or services) are injected, or passed by reference, into a
dependent object (or client) and are made part of the client's state. The pattern separates the creation of a client's dependencies from its own
behavior, which allows program designs to be loosely coupled and to follow the dependency inversion and single responsibility principles.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class SomeClass {

  public function __construct() {
    $this-&gt;object = new SomeOtherClass();  // Noncompliant
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
class SomeClass {

  public function __construct(SomeOtherClass $object) {
    $this-&gt;object = $object;
  }
}
</pre>

