<p>If a function does not return anything, it makes no sense to use its output. Specifically, passing it to another function, or assigning its
"result" to a variable is probably a bug because such functions return nothing, which is probably not what was intended.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$result = closedir($dir_handle); // Noncompliant, "closedir" does not return anything.
</pre>
<h2>Compliant Solution</h2>
<pre>
closedir($dir_handle);
</pre>

