<p>Because it is dynamically typed, PHP does not enforce a return type on a function. This means that different paths through a function can return
different types of values, which can be very confusing to the user and significantly harder to maintain.</p>
<p>In particular, it is consequently also possible to mix empty <code>return</code> statements (implicitly returning <code>null</code>) with some
returning an expression. This rule verifies that all the <code>return</code> statements from a function are consistent.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function foo($a) { // Noncompliant, function will return "true" or null
  if ($a == 1) {
    return true;
  }
  return;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function foo($a) {
  if ($a == 1) {
    return true;
  }
  return false;
}
</pre>
<p>or </p>
<pre>
function foo($a) {
  if ($a == 1) {
    return true;
  }
  return null;
}
</pre>

